File: easy/excel_sheet_column/excel_sheet_column.py
class Solution(object):
    def convertToTitle(self, n):
        """
        :type n: int
        :rtype: str
        """
        column_name = ""
        while n > 0:
            letter_number = n % 26
            if letter_number == 0:
                letter_number = 26
                n = n - 26
            letter = ord('A') + letter_number - 1
            column_name = chr(letter) + column_name
            n = int(n / 26)

        return column_name


s = Solution()
print("Solution 1 :", s.convertToTitle(1))
print("Solution 2 :", s.convertToTitle(28))
print("Solution 3 :", s.convertToTitle(701))
print("Solution 4 :", s.convertToTitle(52))
print("Solution 5 :", s.convertToTitle(703))

File: easy/excel_sheet_column/excel_sheet_column.go
package main

import (
	"fmt"
)

func convertToTitle(n int) string {
	columnName := ""

	for n > 0 {
		letterNumber := n % 26
		if letterNumber == 0 {
			columnName = "Z" + columnName
			n = n - 26
		} else {
			letter := rune(int('A') + letterNumber - 1)
			columnName = string(letter) + columnName
		}
		n = n / 26
	}

	return columnName
}

func main() {
	fmt.Println("Solution 1 : ", convertToTitle(1))
	fmt.Println("Solution 2 : ", convertToTitle(28))
	fmt.Println("Solution 3 : ", convertToTitle(701))
	fmt.Println("Solution 4 : ", convertToTitle(52))
	fmt.Println("Solution 5 : ", convertToTitle(703))
}
